be/src/vec/functions/function_has_all.h

```diff
@@ -0,0 +1,231 @@
+                res=true;
+                pos_1=offset_1;
+                pos_2=offset_2;
+            }
+    }
+    template <typename NestedColumnType>
+    void execute_has_all_num(const IColumn& src1_column,const IColumn& src2_column,const NullMapType* src1_null_map,
+        const NullMapType* src2_null_map,ColumnUInt8::Container& result,const ColumnArray::Offsets64& offsets1,const ColumnArray::Offsets64& offsets2){
+        using NestType = typename NestedColumnType::value_type;
+        const NestedColumnType* src1_data_concrete = reinterpret_cast<const NestedColumnType*>(&src1_column);
+        if (!src1_data_concrete) {
+            return;
+        }
+        const NestedColumnType* src2_data_concrete = reinterpret_cast<const NestedColumnType*>(&src2_column);
+        if (!src2_data_concrete) {
+            return;
+        }
+        const PaddedPODArray<NestType>& src1_datas = src1_data_concrete->get_data();
+        const PaddedPODArray<NestType>& src2_datas = src2_data_concrete->get_data();
+        size_t pos_1 = 0;
+        size_t pos_2 = 0;
+        for (auto m=0;m<offsets1.size();++m){
+            auto offset_1=offsets1[m];
+            auto offset_2=offsets2[m];
+            bool res=true;
+            //check src1_null_map is or not has true
+            bool src1_null_flag=false;
+            for(size_t n = pos_1; n < offset_1; ++n){
+                if(src1_null_map && (*src1_null_map)[n]){
+                    src1_null_flag=true;
+                    break;
+                }
+            }
+            for (size_t i = pos_2; i < offset_2; ++i){
+                for (size_t j = pos_1; j < offset_1; ++j){
+                    if(src2_null_map&&(*src2_null_map)[i]&&src1_null_flag){
+                        break;
+                    }
+                    if(src2_null_map&&(*src2_null_map)[i]&&!src1_null_flag){
+                        res=false;
+                        break;
+                    }
+                    if(src1_datas[j]==src2_datas[i]){
+                            break;
+                        }
+                        else if(j==offset_1-1){
+                            res=false;
+                            break;
+                        }
+                }
+                if(res==false)
+                    break;
+            }
+            result[m]=res;
+            res=true;
+            pos_1=offset_1;
+            pos_2=offset_2;
+        }
+    }
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override{
+        ColumnPtr src1_column =
+                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
+        const auto& src1_column_array = check_and_get_column<ColumnArray>(*src1_column);
+        const auto& offsets1 = src1_column_array->get_offsets();
+        const auto* src1_nested_column = &src1_column_array->get_data();
```

Comment:
rename these variables, eg `lhs_nested_column` is better than `src1_nested_column`